apps/design/frontend/src/api.ts

```diff
@@ -153,10 +154,23 @@ export const exportAllBallots = {
   },
 } as const;
 
-export const exportBallot = {
-  useMutation() {
+interface GetBallotPreviewInput {
+  electionId: Id;
+  precinctId: string;
+  ballotStyleId: string;
+  ballotType: BallotType;
+  ballotMode: BallotMode;
+}
+
+export const getBallotPreviewPdf = {
```

Comment:
This doesn't seem to handle unexpected errors well. If the backend returns a 500 this just keeps retrying while the preview area shows a spinner. There's no indication that something has gone wrong.

Reply:
Good callout, addressed this here: https://github.com/votingworks/vxsuite/pull/4564